src/main/scala/calc/NodeCompiler.scala

```diff
@@ -0,0 +1,31 @@
+package calc
+
+import org.scalajs.core.ir.{Trees => irt}
+
+object NodeCompiler {
+  def apply(f: Tree => irt.Tree): NodeCompiler = {
```

Comment:
Yes, I agree that a simple pattern matching will be more efficient and sufficient for this simple language, what I wanted to show you here is my attempt to consider about the modularity of the compiler, i.e when the rules become large, a single pattern matching will be unmaintainable, I think.
I saw this approach of breaking compilers into smaller "node compilers" in CoffeeScript compiler.

I'll change it to use pattern matching, thanks for the feedback